src/cpp/src/utils/paged_attention_transformations.cpp

```diff
@@ -78,6 +83,19 @@ void apply_paged_attention_transformations(std::shared_ptr<ov::Model> model, Dev
     set_kv_cache_type_and_shape(model, device_config);
 }
 
+void apply_gather_before_matmul_transformation(std::shared_ptr<ov::Model> model) {
```

Comment:
to rename fie to `transformations.hpp` or `apply_transformation.hpp`?

Reply:
in general, `apply_gather_before_matmul_transformation` does not have strong relationship to PA and it can be applied to stateful models as well.

maybe we can move to common utils.hpp ? and avoid any renaming